input/types/types.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"net"
+	"time"
+)
+
+// Builder specifes the interface any delimiter and parser must implement.
```

Comment:
I had my problems with finding a name for this, but I think the term `Tokenizer` will work.